### PR TITLE
[MISC] Update snap and renovate fixes

### PR DESCRIPTION
**Pull request by dragomirp**
Update snap

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -51,10 +51,10 @@ jobs:
       fail-fast: false
       matrix:
         juju:
-          - agent: 2.9.46
-            libjuju: ^2
+          - agent: 2.9.46  # renovate: latest juju 2
```
Comment (dragomirp) [inline .github/workflows/ci.yaml:54]: The backup tests fail on 2.9.47. Charm cannot be redeployed with the same name.

File: src/charm.py [hunk 2]
```diff
@@ -1390,7 +1390,11 @@ def push_tls_files_to_workload(self) -> bool:
         if cert is not None:
             self._patroni.render_file(f"{PATRONI_CONF_PATH}/{TLS_CERT_FILE}", cert, 0o600)
 
-        return self.update_config()
+        try:
+            return self.update_config()
+        except Exception:
+            logger.exception("TLS files failed to push. Error in config update")
+            return False
```
Comment (dragomirp) [inline src/charm.py:1397]: We don't really check for exceptions or return value of `update_config()` in most places, we should most probably go over and handle such cases.

File: tests/integration/test_backups.py [hunk 3]
```diff
@@ -355,12 +352,9 @@ async def test_restore_on_new_cluster(ops_test: OpsTest, github_secrets) -> None
...
-        await wait_for_idle_on_blocked(
-            ops_test,
-            database_app_name,
-            0,
-            S3_INTEGRATOR_APP_NAME,
-            ANOTHER_CLUSTER_REPOSITORY_ERROR_MESSAGE,
+        unit = ops_test.model.units.get(f"{database_app_name}/0")
+        await ops_test.model.block_until(
+            lambda: unit.workload_status_message == ANOTHER_CLUSTER_REPOSITORY_ERROR_MESSAGE
         )
```
Comment (dragomirp) [inline tests/integration/test_backups.py:358]: Model fails to idle with the new snap. I tried with one minute ffwd interval, but it didn't help.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (dragomirp) [reply .github/workflows/ci.yaml:54]: This seems to be [fixed](https://github.com/canonical/postgresql-operator/actions/runs/8615188370/job/23610463141?pr=433) with juju 2.9.49, but there are issues with the subordinates test. I will update in a separate PR.

**Review by marceloneppel — APPROVED** [review]
LGTM!

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (taurus-forever) [reply .github/workflows/ci.yaml:54]: I cannot find such fixes mentioned in Juju 2.9.47+ release notes: https://discourse.charmhub.io/t/roadmap-releases/5064 
Maybe [this](https://bugs.launchpad.net/juju/+bug/2058506)... strange... Will ask Juju Team to improve release notes as 2.9.48 is missing there completely.

File: src/charm.py (hunk 2 above)
Comment (taurus-forever) [reply src/charm.py:1397]: Yeah.... this is what we saw with @beliaev-maksim yesterday on a troubleshooting. Tnx for the quick fix here!

**Review by taurus-forever — APPROVED** [review]
Thank you!

**Pull request merged**